firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/UpdateApkClient.java

```diff
@@ -57,51 +59,47 @@ public void updateApk(
       @NonNull String downloadUrl,
       @NonNull Activity currentActivity) {
 
-    downloadApk(downloadUrl, updateTask)
+    this.cachedUpdateTask = updateTask;
```

Comment:
I'm realizing now that this might make testing a little more difficult. If we want to test a method other than updateApp, we'll need to make sure this is non-null

Reply:
True, maybe we could add a method like 
`@VisibleForTesting
  void setCachedUpdateTask(UpdateTaskImpl updateTask) {
    this.cachedUpdateTask = updateTask;
  }`  or just go back to propagating the taks

Reply:
Both seem reasonable to me :) I'd say go with whatever makes most sense to you